### PR TITLE
ecp5: Reduce some chipdb fields from 32 to 16 bit

### PR DESCRIPTION
**Pull request by mkj**
This reduces the final binary size by ~9 MB for 85k

### CODE REVIEW

**Comment by gatecat** [comment]
Some of these look alright, others are causing things to no longer be suitably aligned hence the failure in CI.

**Comment by mkj** [comment]
Not sure why building locally works OK but CI fails, I've reordered some fields now.

File: ecp5/arch.h [hunk 1]
```diff
@@ -54,24 +54,24 @@ NPNR_PACKED_STRUCT(struct BelPortPOD {
...
+    int16_t timing_class;
     int16_t tile_type;
     int8_t pip_type;
     int8_t padding_0;
 });
 
 NPNR_PACKED_STRUCT(struct PipLocatorPOD {
     LocationPOD rel_loc;
-    int32_t index;
+    int16_t index;
```
Comment (gatecat) [inline ecp5/arch.h:66]: I'd rather keep `PipLocatorPOD` as being 32-bit aligned

**Comment by mkj** [comment]
The structs all keep 32 bit alignment now. Turns out I was missing `-DNDEBUG` in my local test build here.

**Review by gatecat — APPROVED** [review]
LGTM now, thanks!

**Pull request merged**